crates/shim/src/cgroup.rs

```diff
@@ -102,15 +115,194 @@ fn write_process_oom_score(pid: u32, score: i64) -> Result<()> {
         .map_err(io_error!(e, "write oom score"))
 }
 
-/// Collect process cgroup stats, return only necessary parts of it
 #[cfg_attr(feature = "tracing", tracing::instrument(level = "info"))]
 pub fn collect_metrics(pid: u32) -> Result<Metrics> {
+    let cgroup = get_cgroup(pid)?;
+    if cgroup.v2() {
+        collect_metrics_v2(pid)
+    } else {
+        collect_metrics_v1(pid)
+    }
+}
+
+/// Collect process cgroup stats, return only necessary parts of it
+#[cfg_attr(feature = "tracing", tracing::instrument(level = "info"))]
+pub fn collect_metrics_v1(pid: u32) -> Result<Metrics> {
     let mut metrics = Metrics::new();
 
     let cgroup = get_cgroup(pid)?;
 
     // to make it easy, fill the necessary metrics only.
-    for sub_system in Cgroup::subsystems(&cgroup) {
+    for sub_system in cgroup.subsystems() {
+        match sub_system {
+            Subsystem::Cpu(cpu_ctr) => {
+                let mut throttle = Throttle::new();
+                let stat = cpu_ctr.cpu().stat;
+                for line in stat.lines() {
+                    let parts = line.split(' ').collect::<Vec<&str>>();
+                    if parts.len() != 2 {
+                        Err(Error::Other(format!("invalid cpu stat line: {}", line)))?;
+                    }
+
+                    // https://github.com/opencontainers/runc/blob/dbe8434359ca35af1c1e10df42b1f4391c1e1010/libcontainer/cgroups/fs2/cpu.go#L70
+                    match parts[0] {
+                        "nr_periods" => {
+                            throttle.set_periods(parts[1].parse::<u64>().unwrap());
```

Comment:
Could you please use `unwrap_or_default()` to avoiding panic directly?

Reply:
Done.